src/transformers/generation/utils.py

```diff
@@ -1334,6 +1327,58 @@ def _get_static_cache(self, max_batch_size: int, max_cache_len: int) -> StaticCa
             self._static_cache.reset()  # reset the cache for a new generation
         return self._static_cache
 
+    def _prepare_special_tokens(
+        self, generation_config: GenerationConfig, kwargs_has_attention_mask: Optional[bool] = None
+    ) -> Tuple[Optional[torch.Tensor]]:
+        """
+        Prepares the special tokens for generation, overwriting the generation config with their processed versions
+        converted to tensor.
+
+        Note that `generation_config` is changed in place and stops being serializable after this method is called.
+        That is no problem is callen within `generate` (`generation_config` is a local copy that doesn't leave the
+        function). However, if called outside `generate`, consider creating a copy of `generation_config` first.
+        """
+
+        # Convert special tokens to tensors (if they exist)
```

Comment:
It feels like it would be easier to do this when we load them in the generation config, and then when we save them we call `tolist()`? no? 

One thing is that `_tensor_or_none` seems to be used elsewhere a lot, even if it's potentially breaking, (deprecation cycle?) having tensor all the time should help no? This way there is no " preparation" for the decoder input ids if they are already saved in the correct format or something like that

Reply:
uhmmm let me give it a go, to see if something breaks

Reply:
😢 I see.